### ISSUE TITLE
Project sunsetting

### ISSUE THREAD

**Issue by dnephin** [dnephin/pre-commit-golang#98]
Hello everyone,

Unfortunately I no longer have the time or motivation to maintain this project. I have not used `pre-commit` or these hooks in a few years. I never added tests to the project, so it's difficult for me to review and approval pull requests with any confidence. There have been a number of PRs that have broken things for other people. There are still plenty of bugs to fix, and there are frequent requests to publish new releases.

There are already other projects offering pre-commit hooks for Go. You can find a list here: https://pre-commit.com/hooks.html. Some of those seem to offer a lot more than this project.

If someone who's been active on this project is really keen on maintaining these hooks, I can link to your fork from the README to redirect people to the new fork. Otherwise I will likely be archiving this project sometime in the next year.

I'd like to thank everyone who has contributed to this project. I hope it was a reasonably pleasant experience.

**Comment by dnephin** [comment]
Yes, I don't really use git hooks anymore. I'll run tests on every file save, and my IDE mostly takes care of formatting and import order every time a file is save. I might do a full test run and lint before committing, but I run it manually. CI catches anything I missed, but it happens infrequently enough.

**Comment by trallnag** [comment]
ok, thanks for the insight

**Comment by guettli** [comment]
@dnephin "tests on every save" This sound interesting. Could you please share how you do this?

**Comment by dnephin** [comment]
Ya, I get a ton of value from running tests on every save!

I use `gotestsum --watch -ftestname -- -count=1`. Sometimes if I'm working in a large project I'll add `-short` or `-run=<name>` to that command to run a subset of tests in the package.

More details about that here: https://github.com/gotestyourself/gotestsum#run-tests-when-a-file-is-saved

By default this runs only the tests in the package that changed (because generally those are the relevant tests), but if your test suite is fast enough and you want to run every test, you could also use `./...` at the end of the command to run everything.

**Comment by tonsV2** [comment]
Maybe archive and refer to [this](https://github.com/TekWizely/pre-commit-golang) project instead?

**Comment by bhundven** [comment]
So, I've ran into this project and https://github.com/TekWizely/pre-commit-golang, and decided to copy/modify some of the pre-commit checks out of this repo here: https://github.com/bhundven/golang-pre-commit

I'm open to PRs and issues, as there may be some initial issues :) . Not looking for funding. Enjoy!

**Comment by davidaparicio** [comment]
> Maybe archive and refer to [this](https://github.com/TekWizely/pre-commit-golang) project instead?

The @TekWizely's project you refer to seems similarly abandoned 😉 
Thanks a lot @bhundven for your initiative ❤

**Comment by TekWizely** [comment]
> > Maybe archive and refer to [this](https://github.com/TekWizely/pre-commit-golang) project instead?
> 
> The @TekWizely's project you refer to seems similarly abandoned 😉 
> Thanks a lot @bhundven for your initiative ❤ 

I can assure you that TekWizely's project is not abandoned - its just mature and doesn't need a lot of attention - also with the addition of the generic invokers i dont need a relase for every new tool that comes out.

**Comment by TekWizely** [comment]
> So, I've ran into this project and https://github.com/TekWizely/pre-commit-golang, and decided to copy/modify some of the pre-commit checks out of this repo here: https://github.com/bhundven/golang-pre-commit
> 
> I'm open to PRs and issues, as there may be some initial issues :) . Not looking for funding. Enjoy!

what enhancements are you making?

I'm not aware of any deficiencies in my project but i know Go has had a lot of changes in the last few releases.

**Comment by davidaparicio** [comment]
> > > Maybe archive and refer to [this](https://github.com/TekWizely/pre-commit-golang) project instead?
> > 
> > 
> > The [@TekWizely](https://github.com/TekWizely)'s project you refer to seems similarly abandoned 😉
> > Thanks a lot [@bhundven](https://github.com/bhundven) for your initiative ❤
> 
> I can assure you that TekWizely's project is not abandoned - its just mature and doesn't need a lot of attention - also with the addition of the generic invokers i dont need a relase for every new tool that comes out.

I said this because I saw that the last release (1.0.0 RC1) has been out since October 2022.
For me, a release candidate is not a stable version and is not ready for a production environment.

I usually assess the maturity of projects before using them (to avoid potential issues later) by considering multiple factors, including their versioning, the date of the last commit, and the community's enthusiasm, measured by the number of GitHub stars.
